src/encoding.ts

```diff
@@ -429,15 +432,16 @@ export function extractTransformsFromEncoding(oldEncoding: Encoding<any>, config
             }
           }
         }
+
         // now the field should refer to post-transformed field instead
-        encoding[channel] = newFieldDef;
+        encoding[channel as any] = newFieldDef;
```

Comment:
need this to resolve: Expression produces a union type that is too complex to represent.ts(2590)